platzky/plugin/plugin_loader.py

```
@@ -1,21 +1,109 @@
+"""Plugin discovery, loading, and lifecycle management."""
+
+from __future__ import annotations
+
 import importlib
+import importlib.metadata
 import inspect
 import logging
-import os
 from types import ModuleType
-from typing import Any, Optional, Type
+from typing import TYPE_CHECKING, Any, Optional, Type
```

Ruff (0.15.12)

[warning] `typing.Type` is deprecated, use `type` instead

(UP035)